include/windows/osd.h

```diff
@@ -0,0 +1,168 @@
+
+#ifndef _WINDOWS_OSD_H_
+#define _WINDOWS_OSD_H_
+
+#include <winsock2.h>
+#include <ws2def.h>
+#include <windows.h>
+#include <assert.h>
+#include <inttypes.h>
+
+#include <time.h>
+
+struct iovec
+{
+    void *iov_base; /* Pointer to data.  */
+    size_t iov_len; /* Length of data.  */
+};
+
+#define strncasecmp _strnicmp
+#define SHUT_RDWR SD_BOTH
+#define CLOCK_MONOTONIC 1
+
+#ifndef EAI_SYSTEM
+# define EAI_SYSTEM   -11
+#endif
+
+typedef int pid_t;
+
+static int clock_gettime(int which_clock, struct timespec* tp)
+{
+    LARGE_INTEGER freq;
+    LARGE_INTEGER count;
+
+    QueryPerformanceFrequency(&freq);
+    QueryPerformanceCounter(&count);
+
+    tp->tv_sec = (time_t)((double)count.QuadPart / freq.QuadPart);
+    tp->tv_nsec = (long)((uint64_t)((double)count.QuadPart * 1000000 / freq.QuadPart) % 1000000);
```

Comment:
Please watch line length, and try to keep to around 80 characters based on readability.